### ISSUE TITLE
Sending a message that is over 160 characters causes an error

### ISSUE THREAD

**Issue by gnu-user** [tinfoilhat/tinfoil-sms#9]
## Issue

Sending a message that is over 160 causes an error, (this is how I discovered the nonce issue), see the error log below for more details of the error.
## Proposed Solution

I did some googling and the solution seems to be to use the method android.telephony.sendMultipartTextMessage to send text messages that are more than 160 characters.

The method description states that it _sends a multi-part text based SMS. The callee should have already divided the message into correctly sized parts by calling divideMessage._

Link to method documentation: [http://bit.ly/KlQXhi](http://bit.ly/KlQXhi)
## Additional info

For cryptographic security, could also include the message part numbering, but this is nearly impossible as the cipher used is a stream cipher, and if they're already reordering your messages maliciously (since GSM has spec for properly ordering multi-part messages) then including the ordering won't help at all (since they could just as easily block your messages/corrupt them anyways).
## Logcat Output

```
05-12 19:08:27.655: W/System.err(287): java.lang.NullPointerException
05-12 19:08:27.665: W/System.err(287):  at android.os.Parcel.readException(Parcel.java:1253)
05-12 19:08:27.665: W/System.err(287):  at android.os.Parcel.readException(Parcel.java:1235)
05-12 19:08:27.665: W/System.err(287):  at com.android.internal.telephony.ISms$Stub$Proxy.sendText(ISms.java:369)
05-12 19:08:27.665: W/System.err(287):  at android.telephony.SmsManager.sendTextMessage(SmsManager.java:87)
05-12 19:08:27.665: W/System.err(287):  at com.tinfoil.sms.utility.SMSUtility.sendSMS(SMSUtility.java:159)
05-12 19:08:27.665: W/System.err(287):  at com.tinfoil.sms.utility.SMSUtility.sendMessage(SMSUtility.java:294)
05-12 19:08:27.665: W/System.err(287):  at com.tinfoil.sms.messageQueue.MessageSender.run(MessageSender.java:112)
05-12 19:08:27.665: W/System.err(287):  at java.lang.Thread.run(Thread.java:1096)
```

**Comment by dataBaseError** [comment]
The reason for the error was a limit was placed on the amount of text a message could accept. Support for multi-part messages has now been added so now those restrictions have been removed.

**Comment by Xoib** [comment]
By the way even with sendMultipartTextMessage we have this exception with mightytext : 

E/texty   ( 5667): onHandleIntent= java.lang.NullPointerException
E/texty   ( 5667): java.lang.NullPointerException
E/texty   ( 5667):  at android.os.Parcel.readException(Parcel.java:1431)
E/texty   ( 5667):  at android.os.Parcel.readException(Parcel.java:1379)
E/texty   ( 5667):  at com.android.internal.telephony.ISms$Stub$Proxy.sendMultipartText(ISms.java:508)
E/texty   ( 5667):  at android.telephony.SmsManager.sendMultipartTextMessage(SmsManager.java:157)
E/texty   ( 5667):  at com.texty.sms.common.n.a(Unknown Source)
E/texty   ( 5667):  at com.texty.sms.C2DMReceiver.a(Unknown Source)
E/texty   ( 5667):  at com.texty.sms.C2DMReceiver.a(Unknown Source)
E/texty   ( 5667):  at com.google.android.c2dm.a.onHandleIntent(Unknown Source)
E/texty   ( 5667):  at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/texty   ( 5667):  at android.os.Handler.dispatchMessage(Handler.java:99)
E/texty   ( 5667):  at android.os.Looper.loop(Looper.java:137)
E/texty   ( 5667):  at android.os.HandlerThread.run(HandlerThread.java:60)

**Comment by gnu-user** [comment]
Hi Xoib,

Thank you for letting us now, I have re-opened the issue if we find a solution it will be posted here and the issue will be closed.

By any chance, are you testing the application using the android emulator? We have had a number of issues with testing SMS using the android emulators and found out that they are known issues with the android emulator that have not been resolved (even testing with the latest version as of this date).

**Comment by dataBaseError** [comment]
This issue was fixed in 189b3a7ea2b35fc9e640d7cfcb345be2c9b2a504
The last problem encountered was a slight oversight by only using the first part of the message rather then taking the concatenated message.
